### PR TITLE
docs: review integrations bedrock

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed Changes**
- docs: review bedrock
- refactor: drop duplicate code
- fix https://github.com/deepset-ai/haystack/issues/7180

### CODE REVIEW

File: integrations/amazon_bedrock/src/haystack_integrations/common/amazon_bedrock/errors.py [hunk 1]
```diff
@@ -1,6 +1,3 @@
-from typing import Optional
```
Comment (wochinge) [inline integrations/amazon_bedrock/src/haystack_integrations/common/amazon_bedrock/errors.py:1]: that's all just duplicate code. Simple inheriting is enough

File: integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py [hunk 2]
```diff
@@ -218,6 +219,13 @@ def _embed_titan(self, documents: List[Document]) -> List[Document]:
 
     @component.output_types(documents=List[Document])
     def run(self, documents: List[Document]):
+        """Embed the provided `Document`s using the specified model.
+
+        :param documents: The `Document`s to embed.
+        :returns: A dictionary with the following keys
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/embedders/amazon_bedrock/document_embedder.py:225]:
```suggestion
        :returns: A dictionary with the following keys:
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py [hunk 3]
```diff
@@ -103,6 +103,10 @@ def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
         Prepares the body for the Claude model
 
         :param prompt: The prompt to be sent to the model.
+        :param inference_kwargs: Additional keyword arguments passed to the handler.
+        :returns: A dictionary with the following keys:
+            - prompt: The prompt to be sent to the model.
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py:108]:
```suggestion
            - `prompt`: The prompt to be sent to the model.
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py [hunk 4]
```diff
@@ -146,7 +150,9 @@ def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
 
         :param prompt: The prompt to be sent to the model.
         :param inference_kwargs: Additional keyword arguments passed to the handler.
-        :returns: A dictionary containing the body for the request.
+        :returns: A dictionary with the following keys:
+            - prompt: The prompt to be sent to the model.
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py:154]:
```suggestion
            - `prompt`: The prompt to be sent to the model.
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py [hunk 5]
```diff
@@ -191,6 +197,14 @@ class AI21LabsJurassic2Adapter(BedrockModelAdapter):
     """
 
     def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
+        """Prepares the body for the Jurassic 2 model.
+
+        :param prompt: The prompt to be sent to the model.
+        :param inference_kwargs: Additional keyword arguments passed to the handler.
+        :returns: A dictionary with the following keys:
+            - prompt: The prompt to be sent to the model.
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py:205]:
```suggestion
            -  `prompt`: The prompt to be sent to the model.
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py [hunk 6]
```diff
@@ -226,7 +240,9 @@ def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
 
         :param prompt: The prompt to be sent to the model.
         :param inference_kwargs: Additional keyword arguments passed to the handler.
-        :returns: A dictionary containing the body for the request.
+        :returns: A dictionary with the following keys
+            - inputText: The prompt to be sent to the model.
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py:244]:
```suggestion
            - `inputText`: The prompt to be sent to the model.
```

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py [hunk 7]
```diff
@@ -270,7 +286,9 @@ def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
 
         :param prompt: The prompt to be sent to the model.
         :param inference_kwargs: Additional keyword arguments passed to the handler.
-        :returns: A dictionary containing the body for the request.
+        :returns: A dictionary with the following keys:
+            - prompt: The prompt to be sent to the model.
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py:290]:
```suggestion
            - `prompt`: The prompt to be sent to the model.
```

**Review by anakin87 — APPROVED** [review]
LGTM!

I left minor suggestions..

**Pull request merged**